packages/plugin-compiler/src/main/kotlin/io/realm/compiler/RealmModelSyntheticPropertiesGeneration.kt

```diff
@@ -307,8 +307,14 @@ class RealmModelSyntheticPropertiesGeneration(private val pluginContext: IrPlugi
 
                                 val property = value.declaration
                                 val backingField = property.backingField
-                                    ?: error("Property without backing field or type")
-                                val nullable = backingField.type.isNullable()
+                                    ?: error("Property without backing field or type.")
+                                // Nullability applies to the generic type in collections
+                                val nullable = if (value.collectionType == CollectionType.NONE) {
+                                    backingField.type.isNullable()
+                                } else {
+                                    value.genericTypes?.get(0)?.nullable
+                                        ?: error("Missing generic type while processing a collection field.")
```

Comment:
These are good candidates for compiler tests (similar with nullability use cases)

Reply:
I will address this in https://github.com/realm/realm-kotlin/issues/290